include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -1510,6 +1513,158 @@ struct KernelContext {
   OrtKernelContext* ctx_;
 };
 
+/**
+ * Macro that logs a message using the provided logger. Throws an exception if OrtApi::Logger_LogMessage fails.
+ * Example: ORT_CXX_LOG(logger, ORT_LOGGING_LEVEL_INFO, "Log a message");
```

Comment:
Can we make non-formatting logging noexcept?
I think the potential failure can only reasonable occur during formatting.

Reply:
I suppose we can even modify the C API `Logger_LogMessage` to not return a `OrtStatus*` at all. The only potential "errors" would be due to bugs/exceptions in the internal implementation.